This is failing with ```
  File "/tmp/salt-tests-tmpdir/scripts/cli_minion.py", line 27, in <module>
    salt_minion()
  File "/tmp/kitchen/testing/salt/scripts.py", line 181, in salt_minion
    minion.start()
  File "/tmp/kitchen/testing/salt/cli/daemons.py", line 343, in start
    super(Minion, self).start()
  File "/tmp/kitchen/testing/salt/utils/parsers.py", line 1036, in start
    self.prepare()
  File "/tmp/kitchen/testing/salt/cli/daemons.py", line 320, in prepare
    self.minion = salt.minion.MinionManager(self.config)
  File "/tmp/kitchen/testing/salt/minion.py", line 926, in __init__
    install_zmq()
  File "/tmp/kitchen/testing/salt/utils/zeromq.py", line 51, in install_zmq
    zmq.eventloop.ioloop.install()
  File "/usr/lib64/python2.7/site-packages/zmq/eventloop/ioloop.py", line 195, in install
    assert (not ioloop.IOLoop.initialized()) or \
AttributeError: type object 'IOLoop' has no attribute 'initialized'
01:01:14,382 [salt.log.setup                           :1165][ERROR   ] An un-handled exception was caught by salt's global exception handler:
AttributeError: type object 'IOLoop' has no attribute 'initialized'
Traceback (most recent call last):
  File "/tmp/salt-tests-tmpdir/scripts/cli_minion.py", line 27, in <module>
    salt_minion()
  File "/tmp/kitchen/testing/salt/scripts.py", line 181, in salt_minion
    minion.start()
  File "/tmp/kitchen/testing/salt/cli/daemons.py", line 343, in start
    super(Minion, self).start()
  File "/tmp/kitchen/testing/salt/utils/parsers.py", line 1036, in start
    self.prepare()
  File "/tmp/kitchen/testing/salt/cli/daemons.py", line 320, in prepare
    self.minion = salt.minion.MinionManager(self.config)
  File "/tmp/kitchen/testing/salt/minion.py", line 926, in __init__
    install_zmq()
  File "/tmp/kitchen/testing/salt/utils/zeromq.py", line 51, in install_zmq
    zmq.eventloop.ioloop.install()
  File "/usr/lib64/python2.7/site-packages/zmq/eventloop/ioloop.py", line 195, in install
    assert (not ioloop.IOLoop.initialized()) or \
AttributeError: type object 'IOLoop' has no attribute 'initialized'
```

When using pyzmq 17.0.0 and tornado 5.0.2 for the test suite.

--- a/salt/transport/tcp.py
+++ b/salt/transport/tcp.py
@@ -945,7 +945,7 @@ class SaltMessageClient(object):
                 break
             try:
                 kwargs = {}
-                if (self.source_ip or self.source_port):
+                if self.source_ip or self.source_port:
                     if tornado.version_info >= (4, 5):
                         ### source_ip and source_port are supported only in Tornado >= 4.5
                         # See http://www.tornadoweb.org/en/stable/releases/v4.5.0.html

--- a/tests/integration/files/engines/runtests_engine.py
+++ b/tests/integration/files/engines/runtests_engine.py
@@ -21,6 +21,7 @@ import logging
 
 # Import salt libs
 import salt.utils.event
+import salt.utils.async
 
 # Import 3rd-party libs
 from tornado import gen
@@ -69,11 +70,11 @@ class PyTestEngine(object):
         self.sock.bind(('localhost', port))
         # become a server socket
         self.sock.listen(5)
-        netutil.add_accept_handler(
-            self.sock,
-            self.handle_connection,
-            io_loop=self.io_loop,
-        )
+        with salt.utils.async.current_ioloop(self.io_loop):
+            netutil.add_accept_handler(
+                self.sock,
+                self.handle_connection,
+            )
 
     def handle_connection(self, connection, address):
         log.warning('Accepted connection from %s. Role: %s', address, self.opts['__role'])

